feat(multichain-account-service): basic functionality management (#6332)
MetaMask/core: packages/multichain-account-service/CHANGELOG.md

## [Unreleased]

### Added

- Add `setBasicFunctionality` method to control providers state and trigger wallets alignment ([#6332](https://github.com/MetaMask/core/pull/6332))

>   - Add `AccountProviderWrapper` to handle Snap account providers behavior when disabled.
>   - Export `AccountProviderWrapper` from public API to allow external providers to use basic functionality logic

Reviewer: What about?

```suggestion
  - Add `AccountProviderWrapper` to handle Snap account providers behavior according to the basic functionality flag.
```
